Model/Refund/RefundHandler.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Amwal\Payments\Model\Refund;
+
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\CreditmemoRepositoryInterface;
+use Magento\Sales\Model\Order\CreditmemoFactory;
+use Magento\Sales\Model\Order\Creditmemo\ItemFactory;
+use Magento\Sales\Model\Order\Creditmemo;
+
+class RefundHandler
+{
+    protected $creditmemoRepository;
+    protected $creditmemoFactory;
+    protected $creditmemoItemFactory;
+
+    public function __construct(
+        CreditmemoRepositoryInterface $creditmemoRepository,
+        CreditmemoFactory             $creditmemoFactory,
+        ItemFactory                   $creditmemoItemFactory
+    )
+    {
+        $this->creditmemoRepository = $creditmemoRepository;
+        $this->creditmemoFactory = $creditmemoFactory;
+        $this->creditmemoItemFactory = $creditmemoItemFactory;
+    }
+
+
+    public function initiateCreditMemo(OrderInterface $order, array $refundItems, float $refundAmount, float $refundShippingAmount)
+    {
+
+        $refundItems = $this->getItemsToRefund($order, $refundItems);
+        $invoice = $order->getInvoiceCollection()->getFirstItem();
+        $creditMemo = $this->creditmemoFactory->createByInvoice(
+            $invoice,
+            $refundItems
+        );
+
+        foreach ($creditMemo->getAllItems() as $creditmemoItem) {
+            $creditmemoItem->setQty($creditmemoItem->getQty());
+        }
+        $this->creditmemoRepository->save($creditMemo);
+
+        $refundAmountFormatted = $order->getBaseCurrency()->formatTxt($refundAmount);
+
+        $creditMemo->addComment('We refunded ' . $refundAmountFormatted . ' online by Amwal Payments.');
+        $creditMemo->setState(Creditmemo::STATE_REFUNDED);
+        $creditMemo->setBaseGrandTotal($refundAmount);
+        $creditMemo->setGrandTotal($refundAmount);
+
+        foreach ($order->getAllItems() as $orderItem) {
+            if (isset($refundItems['qtys'][$orderItem->getId()]) && $refundItems['qtys'][$orderItem->getId()] > 0) {
+                $orderItem->setQtyRefunded($orderItem->getQtyRefunded() + $refundItems['qtys'][$orderItem->getId()]);
+                $orderItem->setAmountRefunded($orderItem->getAmountRefunded() + $refundAmount);
+            }
+        }
+        $order->setTotalRefunded($order->getTotalRefunded() + $refundAmount);
+        $order->addStatusHistoryComment('We refunded ' . $refundAmountFormatted . ' online by Amwal Payments.');
```

Comment:
Might also be good to make this translatable:
```suggestion
        $order->addStatusHistoryComment(__('We refunded %1 online by Amwal Payments.', $refundAmountFormatted));
```